compiler/rustc_hir_analysis/src/check/method/suggest.rs

```diff
@@ -1201,6 +1203,85 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    fn suggest_range_for_iter(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        source: SelfSource<'tcx>,
+        span: Span,
+        item_name: Ident,
+        ty_str: &str,
+    ) -> bool {
+        if let SelfSource::MethodCall(expr) = source {
+            // avoid suggesting when the method name is not implemented for a `range`
+            let mut search_limit = 4;
+            for (_, parent) in tcx.hir().parent_iter(expr.hir_id) {
+                search_limit -= 1;
+                if search_limit == 0 {
+                    return false;
+                }
+
+                if let Node::Expr(parent_expr) = parent &&
+                    is_range_literal(parent_expr) {
+                    let span_included = match parent_expr.kind {
+                            hir::ExprKind::Struct(_, eps, _) => eps.iter().any(|ep| ep.expr.span.contains(span)),
+                            // `..=` desugars into `::std::ops::RangeInclusive::new(...)`.
+                            hir::ExprKind::Call(ref func, ..) => func.span.contains(span),
+                            _ => false,
+                    };
+
+                    if !span_included {
+                        continue;
+                    }
+                    println!("parent_expr: {:?}", parent_expr);
+
+                    let range_def_id = self.tcx.lang_items().range_struct().unwrap();
+                    let range_ty = self.tcx.type_of(range_def_id);
+                    let call_expr = self
+                                            .tcx
+                                            .hir()
+                                            .expect_expr(self.tcx.hir().get_parent_node(expr.hir_id));
+                    println!("range_ty: {:?}", range_ty);
+                    println!("call_expr: {:?}", call_expr);
+                    println!("item_name: {:?}", item_name);
+                    println!("expr: {:?}", expr);
+                    let pick =  self.lookup_probe(
+                        span,
+                        item_name,
+                        range_ty,
+                        expr,
+                        ProbeScope::TraitsInScope
+                    );
+                    println!("pick: {:?}", pick);
```

Comment:
I tried to `lookup_probe` method such as `rev` in `std::ops::Range<Idx>`, but it's failed.
The code for `lookup_probe` seems complicated, I'm still trying to understand it.

@compiler-errors do you know why it failed? I think `Range` implemented iterator, so it could find the `rev` method.

The log for this part of code is such as:

![image](https://user-images.githubusercontent.com/230646/193241839-b58e1201-fe83-43e8-8187-ddf071246d56.png)

Reply:
I try to suggest user to change 

```console
let _res = 1..10.rev();
```
to 
```console
let _res = (1..10).rev();
```

Reply:
solved it, thanks @lcnr !